packages/template-retail-react-app/app/components/display-price/index.jsx

```diff
@@ -7,61 +7,160 @@
 
 import React from 'react'
 import PropTypes from 'prop-types'
-import {Skeleton, Text} from '@salesforce/retail-react-app/app/components/shared/ui'
+import {Box, Text} from '@salesforce/retail-react-app/app/components/shared/ui'
 import {useIntl} from 'react-intl'
-import {useCurrency} from '@salesforce/retail-react-app/app/hooks'
 
-const DisplayPrice = ({
-    basePrice,
-    discountPrice,
-    isProductASet = false,
-    currency,
-    discountPriceProps,
-    basePriceProps,
-    skeletonProps
-}) => {
+/**
+ * @param priceData - price info extracted from a product
+ * // If a product is a set,
+ *      on PLP, Show From X where X is the lowest of its children
+ *      on PDP, Show From X where X is the lowest of its children and
+ *          the set children will have it own price as From X (cross) Y
+ * // if a product is a master
+ *      on PLP and PDP, show From X (cross) Y , the X and Y are
+ *          sale and list price of variant that has the lowest price (including promotionalPrice)
+ * // if a standard/bundle
+ *      show exact price on PLP and PDP as X (cross) Y
+ * @param currency - currency
+ */
+const DisplayPrice = ({priceData, currency}) => {
```

Comment:
@vmarta @bendvc I found a few limitation when it comes to using formatted message for pricing, it is not possible to inject aria-label into the formatted string for a11y, but you have to do it at the custom tags injection level which destroyed the purpose of using formatted string in the first place. Hence,  I have switched the implementation back to normal react way. 

We can circle back on this one when we figure out our solution with translation.